### ISSUE TITLE
Automatically create supplier_admin role as part of setting up a marketplace

### ISSUE THREAD

**Issue by kenton** [boomerdigital/solidus_marketplace#38]
TL;DR:
Need to do something like this as part of setting up a marketplace store, where role name is "supplier_admin":
https://github.com/solidusio/solidus/blob/master/core/db/default/spree/roles.rb

Background:
We need to use the supplier_admin role as a way of locking down what a supplier admin has access to via CanCan. Using roles allows us to do this without impacting those users that have an admin role, and without having the abilities of an admin role to bleed into what a supplier_admin can do.  For example, previously, whether a user was a supplier_admin was defined by them having an admin role and them having an associated supplier.  That's no good though, because those users would inherit all the roles an admin would have...which isn't what we need.  So we need a new role and an updated AbilityDecorator that uses that role to define what supplier_admins are authorized to do.

**Comment by kenton** [comment]
PR https://github.com/boomerdigital/solidus_marketplace/pull/37
